docs/reference/mapping/types/boolean.asciidoc

```diff
@@ -214,3 +214,39 @@ The following parameters are accepted by `boolean` fields:
 <<mapping-field-meta,`meta`>>::
 
     Metadata about the field.
+
+ifeval::["{release-state}"=="unreleased"]
+[[boolean-synthetic-source]]
+==== Synthetic source
+`boolean` fields support <<synthetic-source,synthetic `_source`>> in their
+default configuration. Synthetic `_source` cannot be used together with
+<<copy-to,`copy_to`>> or with <<doc-values,`doc_values`>> disabled.
+
+Synthetic source always sorts `boolean` fields. For example:
+[source,console,id=synthetic-source-boolean-example]
+----
+PUT idx
+{
+  "mappings": {
+    "_source": { "synthetic": true },
+    "properties": {
+      "bool": { "type": "boolean" }
+    }
+  }
+}
+PUT idx/_doc/1
+{
+  "bool": [true, false, true, false]
+}
+----
+// TEST[s/$/\nGET idx\/_doc\/1?filter_path=_source\n/]
+
+Will become:
+[source,console-result]
+----
+{
+  "bool": [false, false, true, true]
```

Comment:
when I see this I am conflicted: first question that comes to mind is "why use an array of booleans?" but then is users to use this , I guess they may rely on ordering, and JSON arrays do preserve the order, so it looks like supporting retrieval from stored fields could be a good idea to preserve ordering for folks that do care about it?

Reply:
Yeah. I think it's probably right to support loading from stored fields to preserve order. I made a prototype in #87480 and it wasn't as bad as I'd worried it would be. I'm not a fan of the stored field loading infrastructure we have, but it's all good.

But stored fields still do all the other things - they just keep order and dupes. That's *better* for some folks. And it's not too complex to support. So, yeah, I think we should.

And, yeah, and array of booleans is silly. It's more believable in combination with the "I pushed all of the values to leaves" behavior. But it's still not *good*.